### PR TITLE
Fix slot names that start with call

### PR DESCRIPTION
**Pull request by BlakeWilliams**
When a component is registered that starts with `call` the compiler
detects it as a template method which is not correct. This changes the
compiler to look for `call(_|$)` instead of `call` to avoid this
issue.

This means that slots can't (and couldn't) start with `call_` since we
still rely on the `call_` naming convention to generate the
template methods. To make this more dev friendly, this adds a check
that raises an error if a slot name starts with `call_`.

A more long-term fix would be to use some kind of template method
container instead of relying on the `call_` naming convention.

Fixes https://github.com/ViewComponent/view_component/issues/1825

### CODE REVIEW

**Comment by reeganviljoen** [comment]
@BlakeWilliams maybe adding the fact that slot names cant start with `call` should be added to known issues in the docs so users can better troubkle shoot this issue going forward

File: test/sandbox/app/components/header_component.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
```
Comment (camertron) [inline test/sandbox/app/components/header_component.rb:1]: Is this used anywhere?

**Review by camertron — APPROVED** [review]
LGTM! To echo @reeganviljoen's point tho, would this also be a good thing to _call_ out in the docs (see what I did there?!)

**Comment by davidgumberg** [comment]
Minor nit, could add a test that the #1825 case is fixed:

```ruby
def test_slot_names_can_start_with_call
  assert_nothing_raised do
    Class.new(ViewComponent::Base) do
      renders_one :callhome_et
    end
  end
end
```

Otherwise LGTM 👍

**Comment by reeganviljoen** [comment]
@BlakeWilliams any progress on this ?

**Comment by camertron** [comment]
@reeganviljoen I can take this one over and get it across the finish line 😄

**Comment by BlakeWilliams** [comment]
I haven't had a ton of time to dedicate to this recently and documentation is always the hardest part. 🙂 

Thanks for the ping @reeganviljoen and @camertron for getting this shipped! 🚀

**Pull request merged**